ui/src/components/RemoteAccess/PassControlDialog.jsx

```diff
@@ -1,102 +1,62 @@
-import { bindActionCreators } from 'redux';
+import { useDispatch, useSelector } from 'react-redux';
 import { Modal, Button, Form } from 'react-bootstrap';
 import { respondToControlRequest } from '../../actions/remoteAccess';
 
-export class PassControlDialog extends React.Component {
-  constructor(props) {
-    super(props);
-    this.accept = this.accept.bind(this);
-    this.reject = this.reject.bind(this);
-    this.show = this.show.bind(this);
-    this.getObserver = this.getObserver.bind(this);
-  }
-
-  getObserver() {
-    let observer = { name: '', message: '', requestsControl: false };
-
-    for (const o of this.props.remoteAccess.observers) {
-      if (o.requestsControl) {
-        observer = o;
-        break;
-      }
-    }
-
-    return observer;
-  }
+function PassControlDialog() {
+  const dispatch = useDispatch();
 
-  show() {
-    let show = false;
+  const inControl = useSelector((state) => state.login.user.inControl);
+  const requestingObs = useSelector((state) =>
+    state.remoteAccess.observers.find((o) => o.requestsControl),
+  );
 
-    if (this.props.login.user.inControl && this.getObserver().requestsControl) {
-      show = true;
-    }
-
-    return show;
-  }
-
-  accept() {
-    const message = this.message.value;
-    this.props.respondToControlRequest(true, message);
-  }
+  function handleSubmit(evt) {
+    evt.preventDefault();
+    const formData = new FormData(evt.target);
 
-  reject() {
-    const message = this.message.value;
-    this.props.respondToControlRequest(false, message);
+    dispatch(
+      respondToControlRequest(
+        evt.nativeEvent.submitter.name === 'allow',
+        formData.get('message'),
+      ),
+    );
   }
 
-  render() {
-    const observer = this.getObserver();
-
-    return (
-      <Modal show={this.show()} backdrop="static" style={{ zIndex: 10_001 }}>
+  return (
+    <Modal
+      show={inControl && requestingObs}
+      backdrop="static"
+      style={{ zIndex: 10_001 }}
+    >
+      <Form onSubmit={handleSubmit}>
         <Modal.Header>
-          <Modal.Title>{observer.nickname} is asking for control</Modal.Title>
+          <Modal.Title>
+            {requestingObs?.nickname} is asking for control
+          </Modal.Title>
         </Modal.Header>
         <Modal.Body>
-          User "{observer.nickname}" is asking for control:
+          User "{requestingObs?.nickname}" is asking for control:
         </Modal.Body>
         <Modal.Footer>
           <Form.Control
-            ref={(ref) => {
-              this.message = ref;
-            }}
+            name="message"
             defaultValue="Here you go !"
             type="textarea"
             placeholder="Message"
             rows="3"
           />
           <br />
-          <Button size="sm" variant="outline-secondary" onClick={this.accept}>
-            {' '}
-            Give control to "{observer.nickname}"{' '}
+          <Button type="submit" name="allow" variant="success">
```

Comment:
Note that I've added a `Form` element with an `onSubmit` handler. The trick here is that there are two submit buttons that both need access to the form's content. I'm assigning a `name` to each button so I can find which one is used with `evt.nativeElement.submitter.name` in the form's submit handler.